bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/MailBuilder.java

```diff
@@ -122,10 +123,14 @@ public MailBuilder withHtml(String html) {
      *
      * @param urlString the URL as String
      * @return a MailBuilder
-     * @throws MalformedURLException if url has invalid format
+     * @throws IllegalArgumentException if url has invalid format
      */
-    public MailBuilder withURLAttachment(String urlString) throws MalformedURLException {
-        attachmentURLs.add(new URL(urlString));
+    public MailBuilder withURLAttachment(String urlString) throws IllegalArgumentException {
+        try {
+            attachmentURLs.add(URI.create(urlString).toURL());
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("Invalid URL format", e);
```

Comment:
What is the reason for converting this checked exception to a `RuntimeException`?